DMCompiler/Compiler/DM/DMParser.cs

```diff
@@ -1125,8 +1125,10 @@ private DMASTProcStatementIf If() {
 
             BracketWhitespace();
             ConsumeRightParenthesis();
-            Whitespace();
-            Check(TokenType.DM_Colon);
+            if (Check(TokenType.DM_Colon) || Check(TokenType.DM_Period)) {
+                Emit(WarningCode.ExtraToken, loc, "Extra token at end of proc statement");
+            }
```

Comment:
Checking for whitespace first would cause `if(foo)       . += "bar"` to be caught, which is paraphrased from actual SS13 code.

Reply:
This does work though:
```
if(0)     .
    world.log << 1
```

Reply:
`WhitespaceNoNewline()` time?